### PR TITLE
TOOLS: Better support of '-h' argument of ucx_info and ucx_read_profile

### LINKED ISSUE #2632: Usage Documentation: add -h/--help flags for ucx_info, ucx_perftest, ucx_read_profile to print usage
There is no standard `-h` / `--help` flag available for any of the `ucx_*` utilities provided by the UCX package.  The only way to display their usage information currently is to deliberately pass an invalid flag (i.e. `-h`), get an error that the usage was incorrect, have the usage printed, and receive a non-zero exit code.

### LINKED ISSUE #2633: Usage Documentation: ucx_perftest usage text has no mention of a --batch flag
The `ucx_perftest` usage text mentions a short flag `-b` which takes a batchfile as input.  Presumably, this is the short version of the `--batch` flag, but it is not documented.  That said, `/usr/share/doc/ucx/perftest/README` only makes reference to a `--batch` flag, not `-b`.

```
# ucx_perftest --help | grep batch
ucx_perftest: invalid option -- '-'
     -D <layout>[,<layout>]    Data layout. Default is "short" in UCT, "contig" in UCP and previous one in batch mode. Second parameter is for receive side in UCP only.
     -b <batchfile> Batch mode. Read and execute tests from a file.

# grep batch /usr/share/doc/ucx/perftest/README 
This is an example of the "batch" configuration files for ucx_perftest.
ucx_perftest --batch msg_pow2 --batch test_types --batch transports <...>
```

This is a request to have the `ucx_perftest` thus inform the user of both short- and long-versions of flags available for use.

### LINKED ISSUE #2634: Usage Documentation: ucx_info -n usage text does not document arguments required
Currently the `ucx_info` usage text mentions a `-n` flag with no additional arguments passed to it:

```
# ucx_info | grep '\-n'
  -n         Estimated UCP endpoint count (for ucp_init)
```

However, the user will note that `ucx_info -n` does in fact require an argument:

```
# ucx_info -n
ucx_info: option requires an argument -- 'n'
Usage: ucx_info [options]
[...]
```

What is this argument that is required?  This is a request to document it for the user.

### PR DESCRIPTION
**Pull request by yosefe**
Fixes #2633 
Fixes #2634 
Fixes #2632

### CODE REVIEW

**Comment by shamisp** [comment]
Please reference the related issues

**Comment by shamisp** [comment]
Adding @stowellm

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7110/ for details (Mellanox internal link).

**Comment by stowellm** [comment]
Commit f3c3410 looks to address #2632 
Commit c37e5a9 looks to address #2634 
Commit f915e31 looks to address #2633

File: src/tools/profile/read_profile.c [hunk 1]
```diff
@@ -481,12 +494,17 @@ int parse_args(int argc, char **argv, options_t *opts)
             }
             break;
         case 'h':
+            usage();
+            return -127;
```
Comment (stowellm) [inline src/tools/profile/read_profile.c]: Shouldn't `ucx_read_profile -h` return 0 since it successfully displays usage information to the user?

Comment (yosefe) [reply src/tools/profile/read_profile.c]: this is the return value from the function, return value from main() would be 0

**Comment by yosefe** [comment]
@shamisp it's in the commit messages

**Comment by shamisp** [comment]
@yosefe - missed it. Please include such info in PR. Github by default hides the extended commit message.

**Comment by shamisp** [comment]
👍

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7115/ for details (Mellanox internal link).

**Pull request merged**